src/nav/ModalNavBar.js

```diff
@@ -1,33 +1,77 @@
 import React from 'react';
-import { View, Text } from 'react-native';
+import { View, Text, StyleSheet } from 'react-native';
 import { connect } from 'react-redux';
+import Icon from 'react-native-vector-icons/Foundation';
 
 import boundActions from '../boundActions';
 import { CONTROL_SIZE } from '../common/platform';
-import { styles } from '../common';
 import { BRAND_COLOR } from '../common/styles';
+import { styles, Touchable, SearchInput } from '../common';
 import NavButton from './NavButton';
 
-class ModalNavBar extends React.Component {
+const navBarStyles = StyleSheet.create({
+  touchable: {
+    paddingLeft: 10,
+    paddingRight: 10,
+  },
+});
+
 
+class ModalNavBar extends React.Component {
   props: {
     nav: any,
     title: string,
-    popRoute: () => void,
+    isSearchBarApplied: boolean,
+    searchBarOnChange: () => void
+  };
+
+  state: {
+    isSearchEnabled: boolean
+  };
+
+  constructor() {
+    super();
+    this.state = {
+      isSearchEnabled: false
+    };
   }
 
+
+  changeSearchEnableState = () => {
+    this.setState({
+      isSearchEnabled: !this.state.isSearchEnabled
+    });
+  };
+
   render() {
-    const { nav, title, popRoute } = this.props;
+    const { isSearchEnabled } = this.state;
+    const { nav, title, popRoute, searchBarOnChange, isSearchBarApplied } = this.props;
     const textStyle = [styles.navTitle, nav.index > 0 && { marginRight: CONTROL_SIZE }];
 
+    let middleComponent;
+    if (isSearchEnabled) {
+      middleComponent = (
+        <SearchInput
+          onChange={searchBarOnChange}
+          backButtonPress={this.changeSearchEnableState}
+        />
+      );
+    } else {
+      middleComponent = <Text style={textStyle}>{title}</Text>;
+    }
+
     return (
       <View style={styles.navBar}>
-        {nav.index > 0 &&
+        {nav.index > 0 && !isSearchEnabled &&
```

Comment:
This is getting kind of messy because now this component is doing two jobs.

I would suggest creating another component instead called `SearchNavBar` and keeping `ModalNavBar` as is.